core/src/Streamly/Internal/Data/Ring.hs

```diff
@@ -583,17 +531,15 @@ slidingWindowWith n (Fold step1 initial1 extract1 final1) =
     step (Tuple4' rb rh i st) a
         | i < n = do
             rh1 <- liftIO $ unsafeInsert rb rh a
-            liftIO $ touchForeignPtr (ringStart rb)
-            let action = toArray unsafeFoldRingM rb (PTR_NEXT(rh, a))
+            let action = toArray unsafeFoldRingM rb rh1
```

Comment:
BUG: We have perhaps introduced a subtle bug here by not being equivalent to the old code. The old code was doing PTR_NEXT(rh) rather than using rh1 and that was for a good reason I guess. If we are at the last element of the ring, rh1 would become 0 and we would not fold anything.